### PR TITLE
Add: Visualition API for segmentation maps

### LINKED ISSUE #1865: [Visualization] Adding an API to visualize segmentation maps along with images
**Short Description**

It would be great to include an API `keras_cv.visualization.plot_segmentation_map_gallery()` which would help with the visualition of the image along with their segmentation maps.

An exmaple of using this API would be:
Segmentation maps

### PR DESCRIPTION
**Pull request by ariG23498**
# What does this PR do?

Fixes #1865 

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

@ianstenbit

### CODE REVIEW

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        y_pred: (Optional)  a Tensor or NumPy array representing the predicted
+            segmentation masks.
+        kwargs: keyword arguments to propagate to
+            `keras_cv.visualization.plot_image_gallery()`.
+    """
+    assert_matplotlib_installed("plot_segmentation_mask_gallery")
+
+    plotted_images = utils.to_numpy(images)
+
+    # Segmentation maps are of 1 channel, here we repeat the channel 3 times
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:79]: This seems to assume that the masks are not one-hot encoded. A few questions:

- What if the masks are one-hot encoded
- How does this visualization look when num_classes > 1? 

A little demo script demonstrating all of these would be awesome

Comment (ianstenbit) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py:79]: (Perhaps if they are one-hot (because the last dimension is > 1), we should just use argmax to convert them before visualizing

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: keras_cv/visualization/plot_segmentation_mask_gallery.py (hunk 1 above)
Comment (ariG23498) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py:79]: Do you think we should have a `is_one_hot` flag in the parameters? -- This was a great edge case (thanks for this)

Comment (ianstenbit) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py:79]: I'd prefer to infer this, as it should be relatively practical to do so

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    num_classes,
+    y_true=None,
+    y_pred=None,
+    rows=3,
+    cols=3,
+    **kwargs
+):
+    """Plots a gallery of images with corresponding segmentation masks.
+
+    Usage:
```
Comment (jbischof) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:34]: Please move examples section to the end of the docstring. Thank you!

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+from keras_cv import utils
+from keras_cv.utils import assert_matplotlib_installed
+from keras_cv.visualization.plot_image_gallery import plot_image_gallery
+
+
+def build_compatile_segmentation_masks(segmentation_masks):
+    rank = len(segmentation_masks.shape)
+    if rank == 3:
+        # (B, H, W)
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:25]: This could actually be a bit more complicated.

Users could pass un-batched masks with shape `(H, W, num_channels)`

Let's avoid this by stating in the docstring that masks and images must include a batch axis (it's good to include shape info in the docstring anyway)

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    train_ds = train_ds.map(unpackage_tfds_inputs).ragged_batch(16)
+    images, segmentation_masks = next(iter(train_ds.take(1)))
+
+    keras_cv.visualization.plot_segmentation_mask_gallery(
+        images,
+        value_range=(0, 255),
+        num_classes=3, # The number of classes for the oxford iiit pet dataset
+        y_true=segmentation_masks,
+        y_pred=segmentation_masks,
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:81]: For the demo we can probably just show `y_true` instead of copying the same thing into `y_pred`

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+# limitations under the License.
+
+import numpy as np
+
+from keras_cv import utils
+from keras_cv.utils import assert_matplotlib_installed
+from keras_cv.visualization.plot_image_gallery import plot_image_gallery
+
+
+def build_compatile_segmentation_masks(segmentation_masks):
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:22]: maybe `reshape_masks`?

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    rank = len(segmentation_masks.shape)
+    if rank == 3:
+        # (B, H, W)
+        return segmentation_masks[..., np.newaxis]
+    elif rank == 4:
+        # (B, H, W, num_channels) OR (B, H, W, 1)
+        if segmentation_masks.shape[-1] == 1:
+            return segmentation_masks.repeat(repeats=3, axis=-1)
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:30]: This `3` is for RGB, right?

That's worth a code comment because it's a bit confusing as a magic constant

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Nice! Love the example -- I think this is super close

File: keras_cv/visualization/plot_segmentation_mask_gallery.py (hunk 5 above)
Comment (ariG23498) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py:22]: That does make more sense! Thanks for the suggestion. 🤗

File: keras_cv/visualization/plot_segmentation_mask_gallery.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    )
+    ```
+
+    ![Example segmentation mask gallery](https://i.imgur.com/aRkmJ1Q.png)
+    """
+    assert_matplotlib_installed("plot_segmentation_mask_gallery")
+
+    plotted_images = utils.to_numpy(images)
+
+    plotted_y_true = utils.to_numpy(y_true)
```
Comment (ianstenbit) [inline keras_cv/visualization/plot_segmentation_mask_gallery.py:98]: This makes me think `y_true` is required -- should we wrap it in an `if y_true is not None:` (like we do for `y_pred`)

**Review by ianstenbit — APPROVED** [review]
This looks good to me after one small fix -- thank you!

File: keras_cv/visualization/plot_segmentation_mask_gallery.py (hunk 7 above)
Comment (ariG23498) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py:98]: My rationale here is that if we do not keep `y_true` required this API is as good as `plot_image_gallery`. Does that not make the usage redundant?

Comment (ariG23498) [reply keras_cv/visualization/plot_segmentation_mask_gallery.py]: That makes sense to me!

**Review by ianstenbit — APPROVED** [review]
Great -- thank you!
